src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -125,6 +125,10 @@ async def deploy_osd_daemons_for_existing_osds(self, host: str, service_name: st
                 if osd_id in before_osd_uuid_map and osd_id not in replace_osd_ids:
                     # if it exists but is part of the replacement operation, don't skip
                     continue
+                if self.mgr.cache.has_daemon(f'osd.{osd_id}'):
+                    # cephadm daemon instance already exists
+                    logger.debug(f'osd id {osd_id} daemon already exists')
+                    continue
```

Comment:
If you physically move a disk to a different host, you end up here, right? 

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/serve.py#L446-L472